corehq/apps/cloudcare/static/cloudcare/js/form_entry/form_ui.js

```diff
@@ -691,13 +691,24 @@ hqDefine("cloudcare/js/form_entry/form_ui", [
                     element.serverError(null);
                 }
 
-                if (allChildren) {
-                    for (let i = 0; i < allChildren.length; i++) {
-                        if (allChildren[i].control >= constants.CONTROL_IMAGE_CHOOSE) {
-                            allChildren[i].filename = element.answer();
+                const inputControl = [constants.CONTROL_IMAGE_CHOOSE, constants.CONTROL_LABEL,
+                    constants.CONTROL_AUDIO_CAPTURE, constants.CONTROL_VIDEO_CAPTURE];
+
+                let findChildAndSetFilename = function (children) {
+                    for (let child of children) {
+                        if (child.children && child.children.length > 0) {
+                            findChildAndSetFilename(child.children);
+                        } else if (inputControl.includes(child.control) && element.binding() === child.binding &&
```

Comment:
What is this binding property?

Reply:
not a 100% sure what it's used by but as far as I can tell its the "path" to the question. in the simplest instance it'll just be `/data/<question_name>` but it'll always end with the question name

Reply:
interesting... in that case, is there a reason to check equality for both binding and ix? From what i remember, ix is the key used for knockout mapping to identify elements. Did you find that checking for just ix equality isn't strict enough?

Reply:
Yep, at least in the few sample forms I've set up with inputs in different broadcasting/question tile configurations I've found that it does sometimes get a little confused when just checking for `ix`.